packages/flutter/test/cupertino/magnifier_test.dart

```diff
@@ -0,0 +1,256 @@
+  Future<void> showCupertinoMagnifier(
+    BuildContext context,
+    WidgetTester tester,
+    ValueNotifier<MagnifierOverlayInfoBearer> infoBearer,
+  ) async {
+    final Future<void> magnifierShown = magnifierController.show(
+        context: context,
+        builder: (_) => CupertinoTextMagnifier(
+              controller: magnifierController,
+              magnifierOverlayInfoBearer: infoBearer,
+            ));
+
+    WidgetsBinding.instance.scheduleFrame();
+    await tester.pumpAndSettle();
+
+    await magnifierShown;
+  }
+
+  tearDown(() async {
+    if (magnifierController.overlayEntry != null) {
+      magnifierController.overlayEntry!.remove();
+      magnifierController.overlayEntry = null;
+    }
+  });
+
+  group('CupertinoTextEditingMagnifier', () {
+    group('position', () {
+      Offset getMagnifierPosition(WidgetTester tester) {
+        final AnimatedPositioned animatedPositioned =
+            tester.firstWidget(find.byType(AnimatedPositioned));
+        return Offset(
+            animatedPositioned.left ?? 0, animatedPositioned.top ?? 0);
+      }
+
+      testWidgets('should be at gesture position if does not violate any positioning rules', (WidgetTester tester) async {
+        final Key fakeTextFieldKey = UniqueKey();
+        final Key outerKey = UniqueKey();
+
+        await tester.pumpWidget(
+          Container(
+            key: outerKey,
+            color: const Color.fromARGB(255, 0, 255, 179),
+            child: MaterialApp(
+              home: Center(
+                  child: Container(
+                key: fakeTextFieldKey,
+                width: 10,
+                height: 10,
+                color: Colors.red,
+                child: const Placeholder(),
+              )),
+            ),
+          ),
+        );
+        final BuildContext context = tester.element(find.byType(Placeholder));
+
+        // Magnifier should be positioned directly over the red square.
+        final RenderBox tapPointRenderBox = tester.firstRenderObject(find.byKey(fakeTextFieldKey)) as RenderBox;
+        final Rect fakeTextFieldRect = tapPointRenderBox.localToGlobal(Offset.zero) & tapPointRenderBox.size;
+
+        final ValueNotifier<MagnifierOverlayInfoBearer> magnifier = ValueNotifier<MagnifierOverlayInfoBearer>(
+          MagnifierOverlayInfoBearer(
+            currentLineBoundries: fakeTextFieldRect,
+            fieldBounds: fakeTextFieldRect,
+            caratRect: fakeTextFieldRect,
+            // The tap position is dragBelow units below the text field.
+            globalGesturePosition: fakeTextFieldRect.center,
+          ),
+        );
+
+        await showCupertinoMagnifier(context, tester, magnifier);
+
+        // Should show two red squares; original, and one in the magnifier,
+        // directly ontop of one another.
+        await expectLater(
+          find.byKey(outerKey),
+          matchesGoldenFile('cupertino_magnifier.position.default.png'),
```

Comment:
This is probably fine but just double checking: Would there be any benefit to doing this golden on an actual CupertinoTextField instead of this fake one?  If the loupe would be visually unaffected then I say no (then we'd just be adding the risk of the test breaking due to a change related to the field and not the loupe).

Reply:
Talked offline: 

Pros of using textField: The golden looks more natural, and would make more sense at a glance.
Cons: Any breakages or stying changes in textField would lead to this breaking, making lives harder. 

Outcome: gonna stick to naked loupe so that the test hopefully has less false negatives. The test comments are good enough for now if the test starts failing